include/ibamr/INSHierarchyIntegrator.h

```diff
@@ -422,6 +434,12 @@ class INSHierarchyIntegrator : public IBTK::HierarchyIntegrator
      */
     virtual void regridProjection() = 0;
 
+    /*!
+     * Pure virtual method to update the current CFL number (i.e., at the end of
+     * a timestep).
+     */
+    virtual void updateCurrentCFLNumber(const int data_idx, const double dt) = 0;
```

Comment:
I wonder about making this a static that "knows what to do" for different data centerings.

I guess the advantage of making it virtual is if we ever get around to implementing INS schemes that use non-interpolatory kernels.

Note also that a generic implementation could be provided that uses functionality from `ArrayDataNormOps` to do this in a somewhat data-centering agnostic way.

Reply:
We could make it static but I think a good alternative would be to recycle the external version in IBHierarchyIntegrator which can handle both data centerings.

Data is stored in a weird way for side-centered so I don't think using the array operations would make this significantly simpler.